### PR TITLE
[RFC][Doc] Contribution Guideline updates

### PR DESCRIPTION
**Pull request by BarclayII**
## Description
New contribution guideline that *should* lower the bar for third-party contributors... at least a little bit.

In response to @MilkshakeForReal 's ~~sacrifice~~ experience in #964 .  Please check if anything is missing or still not good enough.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] Related issue is referred in this PR

### CODE REVIEW

File: docs/source/contribute.rst [hunk 1]
```diff
@@ -100,8 +100,84 @@ the standard. For example, following variable names are accepted:
...
+To contribute a new model within a specific supported tensor framework (e.g. PyTorch, or MXNet), simply
+
+1. Make a directory with the name of your model (say ``awesome-gnn``) within the directory
+   ``examples/${DGLBACKEND}`` where ``${DGLBACKEND}`` refers to the framework name.
+   
+2. Populate it with your work, and make a pull request once you are done.
+
+   * Instructions for running your program and the results are preferred in your README.
+   
+3. We (the DGL team) will review it, suggesting or making changes as necessary.
```
Comment (jermainewang) [inline docs/source/contribute.rst]: The committers will ...

Comment (BarclayII) [reply docs/source/contribute.rst]: Who are the committers?

File: docs/source/contribute.rst [hunk 2]
```diff
@@ -100,8 +100,84 @@ the standard. For example, following variable names are accepted:
...
+-----------------------------------
+
+To contribute a new model within a specific supported tensor framework (e.g. PyTorch, or MXNet), simply
+
+1. Make a directory with the name of your model (say ``awesome-gnn``) within the directory
+   ``examples/${DGLBACKEND}`` where ``${DGLBACKEND}`` refers to the framework name.
+   
+2. Populate it with your work, and make a pull request once you are done.
+
+   * Instructions for running your program and the results are preferred in your README.
```
Comment (jermainewang) [inline docs/source/contribute.rst]: Populate it with your work, and make a pull request once you are done.
. Please provide a README with at least two sections. First, instructions for running your program. Second, the performance results such as speed or accuracy compared with some alternative implementations.

File: docs/source/contribute.rst [hunk 3]
```diff
@@ -100,8 +100,84 @@ the standard. For example, following variable names are accepted:
...
+
+4. Resolve the suggestions and reviews, and go back to step 3 until approved.
+
+5. Merge it and enjoy your day.
+
+Data hosting
+````````````
+
+One often wishes to upload a dataset when contributing a new runnable model example, especially when covering
+a new field which is not covered in our existing examples.
```
Comment (jermainewang) [inline docs/source/contribute.rst]: a new field not in our existing examples

**Pull request merged**